docs/experimental/extendable-param-files.md

```diff
@@ -0,0 +1,90 @@
+# Using the Extendable Bicep Parameters Feature (Experimental!)
+
+## What is it?
+
+Extendable Bicep Parameter Files is a feature that allows you to extend `.bicepparam` files from another `.bicepparam` file in order to reuse parameters across multiple deployments.
+
+When using extendable bicep parameter files, you will have a main `.bicepparam` file that can be consumed by multiple extended `.bicepparam` files.
+
+## Using
+
+`main.bicep` This is your main bicep file, which will define your parameters for deployment.
+
+```bicep
+param namePrefix string
+param location string
+param symbolicName string
+```
+
+`root.bicepparam` This is your main bicepparam file, which can be reused by multiple extended .bicepparam files and in multiple deployments.
+
+```bicep
+using none
+
+param namePrefix = 'share'
+param location = 'westus'
+```
+
+`leaf.bicepparam` This is your extended bicepparam file, which will refer to one main.bicep file and one main .bicepparam file. Any parameter value in this file will override all previous values.
+
+```bicep
+using 'main.bicep'
+
+extends 'root.bicepparam'
+
+param namePrefix = 'extend'
+param symbolicName = 'test'
+```
+
+Resolved Values
+| Param | Value |
+| -- | -- |
+| namePrefix | `'extend'` |
+| location | `'westus'` |
+| symbolicName | `'test'` |
+
+**Note**: As `foo` is defined in both `root.bicepparam` and `leaf.bicepparam` files, any parameter values in the **`leaf.bicepparam`** file will override the values of the parameter in **both** the `main.bicep` and `root.bicepparam` files.
+
+## Limitations
+
+We do not have support for:
+
+* variables: we will not have variable support in the `root.bicepparam` file to be able to read and override values in other files
+  ```bicep
+  var namePrefix = 'share'
```

Comment:
It's a little odd to have such specific code samples for features that don't work. I think it's ok just to describe the limitation in words.

Reply:
So Anthony had requested I add an explanation and example for each of these bullet points - I'll follow up with him to see if this is what he is looking for or if just a quick description is good enough.

Reply:
Anthony said the examples should be helpful for users - would you be okay if I kept them in?

Reply:
yep - fine with me